uint16 gpio_id
---
uint8 level
bool has_succeeded